File: reddio-java/reddio-api/src/main/java/com/reddio/api/v1/requests/ReddioWithdrawalToApi.java
package com.reddio.api.v1.requests;

import com.reddio.api.v1.rest.*;

import java.util.concurrent.CompletableFuture;

public class ReddioWithdrawalToApi implements SignedReddioApiRequest<WithdrawalToMessage, ResponseWrapper<WithdrawalToResponse>> {

    private ReddioRestClient localRestClient;
    private WithdrawalToMessage request;

    @Override
    public ResponseWrapper<WithdrawalToResponse> send() {
        return null;
    }

    @Override
    public CompletableFuture<ResponseWrapper<WithdrawalToResponse>> sendAsync() {
        return null;
    }

    @Override
    public WithdrawalToMessage getRequest() {
        return this.request;
    }

    @Override
    public Signature getSignature() {
        return this.request.getSignature();
    }
}

File: reddio-java/reddio-api/src/main/kotlin/com/reddio/api/v1/ReddioWithdrawalToApi.kt
package com.reddio.api.v1

import com.reddio.api.v1.requests.SignedReddioApiRequest
import com.reddio.api.v1.rest.*
import java.util.concurrent.CompletableFuture

class ReddioWithdrawalToApi private constructor(
    private val localRestClient: ReddioRestClient,
    private val request: WithdrawalToMessage
) : SignedReddioApiRequest<WithdrawalToMessage?, ResponseWrapper<WithdrawalToResponse?>?> {
    override fun send(): ResponseWrapper<WithdrawalToResponse?>? {
        return null
    }

    override fun sendAsync(): CompletableFuture<ResponseWrapper<WithdrawalToResponse?>?>? {
        return null
    }

    override fun getRequest(): WithdrawalToMessage {
        return request
    }

    override fun getSignature(): Signature {
        return request.getSignature()
    }

    companion object {
        fun build(localRestClient: ReddioRestClient, request: WithdrawalToMessage): ReddioWithdrawalToApi {
            return ReddioWithdrawalToApi(localRestClient, request)
        }
    }
}